### PR TITLE
SMP PT1.12 re-release

### PR DESCRIPTION
**Pull request by YangFei1990**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
SMP PT1.12 re-release
For this release, safety check tools might report a vulnerability for the package commonmarker, which is a dependency of deepspeed. This package is only used to build the documentation pages of deepspeed and won’t be used in the package that gets installed into the DLC. This security issue can be safely ignored and an attempt to upgrade deepspeed version to remediate it might have an inadvertent negative impact on the DLC components functionality.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by tejaschumbalkar** [comment]
Please add related test so validate the functionality of new binary

**Comment by haohanchen-yagao** [comment]
Safety check passed

File: dlc_developer_config.toml [hunk 1]
```diff
@@ -48,7 +48,7 @@ sagemaker_local_tests = false
 # "standard" --> run standard sagemaker remote tests from test/sagemaker_tests
 # "rc" --> run release_candidate_integration tests
 # "efa" --> run efa sagemaker tests
-sagemaker_remote_tests = "off"
```
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:51]: Change it to `efa` in order to run `standard` and `efa` tests

Comment (haohanchen-yagao) [reply dlc_developer_config.toml:51]: Done, changed to efa

**Comment by tejaschumbalkar** [comment]
SM test failures will be fixed by #2164

**Comment by haohanchen-yagao** [comment]
efa test passed, enable rc and local test

File: data/ignore_ids_safety_scan.json [hunk 2]
```diff
@@ -402,7 +402,8 @@
             "py3": {
                 "42772":"for shipping bokeh<=2.3.3 - the last available version for py3.6",
                 "42814":"for shipping bokeh<=2.3.3 - the last available version for py3.6",
-                "42815":"for shipping bokeh<=2.3.3 - the last available version for py3.6"
+                "42815":"for shipping bokeh<=2.3.3 - the last available version for py3.6",
+                "48298":"check PR2145 introduction"
```
Comment (tejaschumbalkar) [inline data/ignore_ids_safety_scan.json:406]: There is no valid description on this PR. Please add the appropriate reason.

File: test/dlc_tests/sanity/test_safety_check.py [hunk 3]
```diff
@@ -430,6 +430,8 @@
                 "42772",
                 "42814",
                 "42815",
+                # check PR2145 introduction
```
Comment (tejaschumbalkar) [inline test/dlc_tests/sanity/test_safety_check.py:433]: this can be removed or substituted for appropriate reasoning

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py [hunk 4]
```diff
@@ -390,6 +391,79 @@ def test_smmodelparallel_mnist_multigpu_multinode_efa(ecr_image, efa_instance_ty
...
+@pytest.mark.parametrize("test_script, num_processes", [("train_gpt_simple.py", 8)])
+@pytest.mark.efa()
+def test_smmodelparallel_gpt2_sdp_multinode_efa(ecr_image, efa_instance_type, sagemaker_regions, test_script, num_processes):
+    """
+    Tests pt gpt2 command via script mode
+    """
+    framework, framework_version = get_framework_and_version_from_tag(ecr_image)
+    if framework == "pytorch" and Version(framework_version) in SpecifierSet("<1.12.0"):
+        pytest.skip("Skipping the test for PT version before 1.12")
+    smp_version = 111
```
Comment (tejaschumbalkar) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py:409]: does this needs to be hardcoded? Would it be the same for new versions for pytorch?

Comment (haohanchen-yagao) [reply test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py:409]: This is intended to maintain the backward compatibility of SMP APIs, for the new PT versions, it could be same unless we introduce new APIs to SMP scripts.

**Comment by tejaschumbalkar** [comment]
SM local test failure on the latest commit

File: data/ignore_ids_safety_scan.json (hunk 2 above)
Comment (haohanchen-yagao) [reply data/ignore_ids_safety_scan.json]: Added description

File: test/dlc_tests/sanity/test_safety_check.py (hunk 3 above)
Comment (haohanchen-yagao) [reply test/dlc_tests/sanity/test_safety_check.py]: Added more detailed reason

**Comment by tejaschumbalkar** [comment]
Pushed a fix to the test script suggested by @ShiboXing

**Review by ShiboXing — DISMISSED** [review]

**Review by tejaschumbalkar — APPROVED** [review]
